src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -76,72 +216,66 @@ public SyntaxNode Mutate(SyntaxNode currentNode)
+                    else if (expressionSyntax is MemberAccessExpressionSyntax memberAccess)
                     {
-                        ifStatement = ifStatement.ReplaceNode(ifStatement.Condition, MutateWithConditionalExpressions(ifStatement.Condition));
+                        actualMutationNode = memberAccess.Expression;
                     }
-                    if (ifStatement.Else != null)
+                    else if (expressionSyntax is AnonymousFunctionExpressionSyntax anonymousFunction)
                     {
-                        ifStatement = ifStatement.ReplaceNode(ifStatement.Else, Mutate(ifStatement.Else));
+                        actualMutationNode = anonymousFunction.Body;
                     }
-                    return ifStatement.ReplaceNode(ifStatement.Statement, Mutate(ifStatement.Statement));
-                }
-                return MutateWithIfStatements(statement);
-            }
-            else
-            {
-                // No statement found yet, search deeper in the tree for statements to mutate
-                var children = currentNode.ChildNodes().ToList();
-                var childCopy = currentNode.TrackNodes(children);
-                foreach (var child in children)
-                {
-                    var mutatedNode = Mutate(child);
-                    var originalNode = childCopy.GetCurrentNode(child);
-                    if (!mutatedNode.IsEquivalentTo(originalNode))
+
+                    if (actualMutationNode != null)
                     {
-                        childCopy = childCopy.ReplaceNode(originalNode, mutatedNode);
+                        var subNodeCopy = expressionSyntax.TrackNodes(actualMutationNode);
+                        var mutant = Mutate(actualMutationNode);
+                        var local = subNodeCopy.GetCurrentNode(actualMutationNode);
+                        var sub = subNodeCopy.ReplaceNode(local, mutant);
+                        currentNodeCopy = currentNodeCopy.ReplaceNode(currentExpressionSyntax, sub);
                     }
+                    else
+                        // The mutations should be placed using a ConditionalExpression
+                        currentNodeCopy = currentNodeCopy.ReplaceNode(currentExpressionSyntax,
+                            MutateWithConditionalExpressions(expressionSyntax));
                 }
-                return childCopy;
+
+                return currentNodeCopy;
             }
+
+            return null;
         }
 
+
         private IEnumerable<Mutant> FindMutants(SyntaxNode current)
+        {
+            foreach (var mutant1 in FlatMutate(current)) yield return mutant1;
+            foreach (var mutant in current.ChildNodes().SelectMany(FindMutants))
+            {
+                yield return mutant;
+            }
+        }
+
+        private IEnumerable<Mutant> FlatMutate(SyntaxNode current)
```

Comment:
I find this name somewhat confusing. Maybe `FindMutants()` and `FindMutantsRecursive()` would make more sence for these two methods.